plugin/fuzz/source/dextool/plugin/fuzz/xml_parse.d

```diff
@@ -0,0 +1,457 @@
+                DataItem data_item = DataItem(elem.tag.attr["name"], elem.tag.attr["type"]);
+                if (auto defaultVal = "defaultValue" in elem.tag.attr) {
+                    data_item.defaultVal = *defaultVal;
+                }
+
+                if (auto startupVal = "startupValue" in elem.tag.attr) {
+                    data_item.startupVal = *startupVal;
+                }
+                cis.data_items.insertBack(data_item);
+
+                break;
+            default:
+                break;
+            }
+        }
+
+        return cis;
+    }
+
+    Types[] types(Document doc, string curr_ns) {
+        if (doc.tag.name == "Types") { //Everything is a type!
+            if (curr_ns.endsWith("::types")) {
+                curr_ns = curr_ns[0..$-("::types").length];
+            }
+            return [getTypes(doc, curr_ns)];
+        } else {
+            return doc.elements.filter!(a => a.tag.name == "Types").map!(a => getTypes(a, curr_ns)).array;
+        }
+    }
+
+    Nullable!Interface_ interfaces(Document doc) {
+        Nullable!Interface_ ret;
+        if (doc.tag.name != "Interface") {
+            return ret;
+        } else {
+            ret = Interface_();
+            foreach (Element elem; doc.elements) {
+                switch (elem.tag.name) {
+                case "ContinousInterface":
+                    ret.ci.insertBack(getContinousInterface(elem));
+                    break;
+                default:
+                    break;
+                }
+            }
+        }
+        return ret;
+    }
+
+    Types merge(Types[] old) {
+        Types new_;
+        foreach (Types old_types; old) {
+            new_.enums = old_types.enums ~ new_.enums;
+            new_.prims = old_types.prims ~ new_.prims;
+            new_.record = old_types.record ~ new_.record;
+            new_.subtypes = old_types.subtypes ~ new_.subtypes;
+        }
+        return new_;
+    }
+
+    Types merge(Types old, ref Types new_) {
+        new_.enums = old.enums ~ new_.enums;
+        new_.prims = old.prims ~ new_.prims;
+        new_.record = old.record ~ new_.record;
+        new_.subtypes = old.subtypes ~ new_.subtypes;
+
+        return new_;
+    }
+
+    Interface_ merge(Interface_ old, Interface_ new_) {
+        new_.ci = old.ci ~ new_.ci;
+
+        return new_;
+    }
+
+public:
+    this(BaseDir bdir) {
+        basedir = bdir;
+        xml_files = (dirEntries(this.basedir, "*.xml", SpanMode.breadth).filter!(a => a.isFile)).map!(
+            a => a.name).array;
+        foundNamespaces = xml_files.map!(a => namespace(a)).filter!(a => a != "").array;
+        nsps = parseBaseDir;
+    }
+
+    Namespace[string] parseBaseDir() {
+        import std.stdio;
+
+        Namespace[string] namespaces;
+        Global glob;
+        foreach (string xml_file; xml_files) {
+            string curr_ns = namespace(xml_file);
+            auto doc = new Document(cast(string) std.file.read(xml_file));
```

Comment:
This may not be the best way, but it is a way. Please be critical and if you find a better way, tell me.
```d
import std.utf : validate;
string raw = () @trusted { return cast(string) std.file.read(xml_file); }();
validate(raw);
```